web/src/beta/features/Editor/Map/InspectorPanel/LayerInspector/FeatureInspector/index.tsx

```diff
@@ -111,6 +111,33 @@
     setFields(fieldArray);
   }, [layer?.sketch?.customPropertySchema, selectedFeature?.properties]);
 
+  const previousFields = useRef<FieldProp[]>([]);
+
+  useEffect(() => {
+    const removedFields = previousFields.current.filter(
+      (field) => !fields.some((f) => f.id === field.id)
+    );
+
+    removedFields.forEach((removedField) => {
+      if (removedField.id && selectedFeature?.properties) {
+        const { [removedField.id]: removed, ...updatedProperties } =
+          selectedFeature.properties;
+        if (!selectedFeature || !sketchFeature?.id || !layer?.id) return;
```

Comment:
Make sense, could you update this? I think you can make the return conditions together at the very top.

Reply:
BTW you can also add a simple comment for what does this new added `previousFields` and `useEffect` work for. It might be a little bit difficult for other developers to get the idea quickly.